src/worker/worker/collectors/base_web_collector.py

```diff
@@ -59,3 +62,52 @@ def xpath_extraction(self, html_content, xpath: str) -> str:
             logger.error(f"No content found for XPath: {xpath}")
             return ""
         return document.xpath(xpath)[0].text_content()
+
+    def extract_meta(self, html_content):
+        html_content = lxml.html.fromstring(html_content)
+        author = ""
+        title = html_content.findtext(".//title", default="")
+
+        if meta_tags := html_content.xpath("//meta[@name='author']"):
+            author = meta_tags[0].get("content", "")
+
+        return author, title
+
+    def parse_web_content(self, web_url, source_id: str, xpath: str = "") -> dict[str, str | datetime.datetime | list]:
+        html_content, published_date = self.html_from_article(web_url)
+        if not html_content:
+            raise ValueError("Website returned no content")
+        author, title = self.extract_meta(html_content)
+
+        if xpath:
+            content = self.xpath_extraction(html_content, xpath)
+        else:
+            # TODO: @with_metadata is deprecated and substituted by @only_with_metadata
+            # TODO: @with_metadata=True more websites return None
+            extract_document = bare_extraction(html_content, with_metadata=True, include_comments=False, url=web_url)
```

Comment:
That sounds like we don't need want this to be true, since we are calling extract_meta anyway, I would set this to false, and remove the lines 91 - 94.

And instead use the result from 80 directly.

Reply:
We might need to spend some time thinking about what is the best way to gather the metadata as the `extract_meta` is not very universal.